lib/crystalball/rspec/prediction_builder.rb

```diff
@@ -39,8 +39,17 @@ def repo
 
       private
 
+      # This method should be overridden in ancestor. Example:
+      #
+      # def base_predictor
+      #   @base_predictor ||= Crystalball::Predictor.new(execution_map, repo, from: config['diff_from'], to: config['diff_to']) do |p|
+      #     p.use Crystalball::Predictor::ModifiedExecutionPaths.new
+      #     p.use Crystalball::Predictor::ModifiedSpecs.new
+      #   end
+      # end
+      #
       def base_predictor
-        @base_predictor ||= config['predictor_class'].new(map, repo, from: config['diff_from'], to: config['diff_to'])
+        raise NotImplementedError, 'Configure `prediction_builder_class_name` in `crystalball.yml`'
```

Comment:
Isn't it an Argument Error in this case?

Reply:
No, it is not implemented. Because users have to implement this method in an ancestor

Reply:
Its too easy to break something by dropping part of required params to predictor. This makes Crystalball literally dependable on user implementation. Consider creating `predictor` method as an extension point
E.g.
```
def predictor
  raise NotImplementedError unless block_given?
  @predictor ||= Crystalball::Predictor.new(map, repo, from: config['diff_from'], to: config['diff_to'], &block)
end
```
so you can use it as 
```
def predictor
  super do |p|
    p.use MyStrategy.new(config['my_param'])
  end
end
```

P.S. yeah lets get rid of `base_` prefix